pkg/controller/stackconfigpolicy/controller.go

```diff
@@ -291,6 +292,27 @@ func (r *ReconcileStackConfigPolicy) doReconcile(ctx context.Context, policy pol
 			return results.WithError(err), status
 		}
 
+		// Copy all the Secrets that are present in spec.elasticsearch.secretMounts
+		if err := reconcileSecreteMounts(ctx, r.Client, es, &policy); err != nil {
+			return results.WithError(err), status
```

Comment:
If you referenced a Secret that doesn't exist in the `secretMounts` you only get an error in the operator log:
```
ERROR   manager.eck-operator    Reconciler error        {
    "service.version": "2.10.0-SNAPSHOT+73898c46", "controller": "stackconfigpolicy-controller", "object": { "name": "scp-policy-invalid-secret", "namespace": "default" }, "namespace": "default", "name": "scp-policy-invalid-secret", "reconcileID": "b6fb61a9-a77f-4fdb-b27a-8cfd9d0d9207",
    "error": "Secret \"nonexistent-secret\" not found","errorCauses": [{ "error": "Secret \"nonexistent-secret\" not found" }]}
```

with an SCP ready configuring 0 resource:

```
NAME                                                                           READY   PHASE   AGE
stackconfigpolicy.stackconfigpolicy.k8s.elastic.co/scp-policy-invalid-secret   0/0     Ready   5m
```

We could better surface this particular error as it requires user intervention to fix:
```suggestion
			if apierrors.IsNotFound(err) {
				err = status.AddPolicyErrorFor(esNsn, policyv1alpha1.ErrorPhase, err.Error())
				if err != nil {
					return results.WithError(err), status
				}
				results.WithResult(defaultRequeue)
			}
			continue
```